build.sh

```diff
@@ -157,7 +157,16 @@ check_uuid_binary(){
 }
 
 run_build() {
-  /home/shippable/cexec/dist/main/main
+  local use_default_arch=true
+  if [ ! -z "$SHIPPABLE_NODE_ARCHITECTURE" ]; then
+    if [ ! -f /home/shippable/cexec/dist/$SHIPPABLE_NODE_ARCHITECTURE/linux/main/main ]; then
+      use_default_arch=false
+      /home/shippable/cexec/dist/$SHIPPABLE_NODE_ARCHITECTURE/linux/main/main
+    fi
+  fi
+  if [ "$use_default_arch" = true ]; then
```

Comment:
make it `==`. i know this works, its for consistency reason we should keep everything same;
also, add an else clause